std/algorithm/searching.d

```diff
@@ -1775,10 +1775,17 @@ if (isRandomAccessRange!R1 && isBidirectionalRange!R2
         && is(typeof(binaryFun!pred(haystack.front, needle.front)) : bool))
 {
     if (needle.empty) return haystack;
-    const needleLength = walkLength(needle.save);
+    static if (hasLength!R2)
+    {
+        immutable needleLength = needle.length;
+    }
+    else
+    {
+        immutable needleLength = walkLength(needle.save);
```

Comment:
This may as well be code written by myself, but it seems to me walking the length of the needle isn't likely to be a great strategy.